app/views/wikis/show.html.erb

```diff
@@ -1,6 +1,26 @@
 <h1><%= @wiki.title %></h1>
-<h1><%= @wiki.body %></h1>
 
-<%= link_to "EDIT WIKI", edit_wiki_path(@wiki), class: 'btn btn-warning' %>
-<%= link_to "NEW WIKI", new_wiki_path(@wiki), class: 'btn btn-success' %>
-<%= link_to "DESTROY", @wiki, method: :delete, class: "btn btn-danger", data: { confirm: 'Are you sure?' } %>
+  <% if (@wiki.private && current_user && current_user!=@wiki.user) ||
```

Comment:
Nice! You could extract this into a helper if you wanted to make this more readable.

Reply:
I think I need to better understand these core functions before moving on to wrap them in a helper.